profiling/src/internal/profile.rs

```diff
@@ -301,9 +299,10 @@ impl Profile {
             encoder.encode(ProfileFunctionsEntry::from(item))?;
         }
 
-        for item in self.strings.into_iter() {
+        for item in self.strings.iter() {
```

Comment:
This is the one case where we don't `into_iter`.  Might be worth a comment explaining why